mvp-spec/partials-updater/src/interpretor.ts

```diff
@@ -0,0 +1,102 @@
+import fs from "fs";
+import { getPartialPath, loadPartial } from "./files";
+import { LexerToken, LexerTokenType, PartialBeginStartKey, PartialBeginEndKey } from "./lexer";
+import * as jq from "node-jq";
+import { fileURLToPath } from "url";
+
+interface PartialConfig {
+    files: string[];
+    block?: string;
+    jq?: string;
+}
+
+/** Interpret the token to generate the content. */
+export async function interpret(tokens: LexerToken[]): Promise<string> {
+    let doc = '';
+    for (let i = 0; i < tokens.length; i++) {
+        const token = tokens[i];
+        switch (token.type) {
+            case LexerTokenType.PlainText: {
+                doc += token.text;
```

Comment:
I guess we would save a lot of memory if using [yield](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Operators/yield) instead of a `doc` variable that holds the entire document.